### ISSUE TITLE
[BUG] wash app deploy should fail or provide indication if component or provider is missing.

### ISSUE THREAD

**Issue by cdmurph32** [wasmCloud/wasmCloud#3034]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [X] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Describe the bug

For complex deployments with multiple components, it can be difficult to determine why a service isn't working. The fact that a component or provider cannot be found should be logged to the wasmcloud.log. Perhaps, `wash app deploy` could fail. 

### Steps to reproduce

1. Attempt to deploy any example with `wash app deploy` without building the component. 

### Expected behavior

I'd expect an error to be logged, and potentially a command failure.

### Environment

- OS: Linux and MacOS
- Shell zsh
- wasmcloud Version v1.2.1

**Comment by brooksmtownsend** [comment]
This is a good idea @cdmurph32, right now we absolutize the paths from relative paths in the manifest and we could go ahead and make sure that exists. At the very least when you deploy the app it should enter a `Failed` state, but we can catch it sooner.

Something to look out for while implementing this is that if it's an absolute path we have to allow that to continue to deploy even if the file doesn't exists, because we could be deploying from a totally different filesystem. Happy to elaborate here

**Comment by cpetig** [comment]
The shell running `wash up` reports
```
2024-11-14T18:36:30.487946Z ERROR wasmcloud_host::wasmbus: failed to scale component component_ref=file:///home/.../hello/build/http_hello_world_s.wasm component_id=rust_hello_world-http_component err=failed to fetch component

Caused by:
    0: failed to read component
    1: No such file or directory (os error 2)
2024-11-14T18:37:00.482627Z ERROR wasmcloud_host::wasmbus: failed to scale component component_ref=file:///home/.../hello/build/http_hello_world_s.wasm component_id=rust_hello_world-http_component err=failed to fetch component
```
while the shell I typed `wash app deploy local.wadm.yaml`
shows `Successfully deployed model rust-hello-world v0.0.1` which clearly is confusing.

**Comment by cpetig** [comment]
deploy_model_from_manifest should check the file name specified in the manifest at `spec/components/properties/image` for existence.

**Comment by vados-cosmonic** [comment]
This should be resolved by #3682 !
